agent/command/exec.go

```diff
@@ -263,13 +280,41 @@ func (c *subprocessExec) getProc(ctx context.Context, taskID string, logger clie
 	return cmd
 }
 
-func addTempDirs(env map[string]string, dir string) {
-	for _, key := range []string{"TMP", "TMPDIR", "TEMP"} {
-		if _, ok := env[key]; ok {
-			continue
+// getExecutablePath returns the absolute path to the command executable to run.
+// If the executable is available in the default runtime environment's PATH,
+// then that path will be used. Otherwise if it can't find the command in the
+// default PATH locations, the command will fall back to checking the command's
+// PATH environment variable for a matching executable location (if any).
+func (c *subprocessExec) getExecutablePath(logger client.LoggerProducer) (absPath string, err error) {
+	cmdPath := c.Env["PATH"]
+	defaultPath, err := exec.LookPath(c.Binary)
+	if defaultPath != "" {
+		return defaultPath, err
+	}
+
+	binaryIsFilePath := strings.Contains(c.Binary, string(filepath.Separator)) || runtime.GOOS == "windows" && strings.Contains(c.Binary, "/")
```

Comment:
Should this instead be `runtime.GOOS != "windows" && strings.Contains(c.Binary, string(filepath.Separator)) || runtime.GOOS == "windows" && strings.Contains(c.Binary, "/")`

Reply:
No, the first case is unconditional because this statement is covering three different valid file path casesthe. For the non-Windows case, the only allowed filepath separator is `/`. However, for Windows, Go accepts either `\` or `/` as a file path separator, so we have to check for either `\` (the native `filepath.Separator`) or `/` (not native to Windows, but still accepted by Go as a file path for Windows) as potential separators.

Reply:
Ah gotcha, thanks for that. Can you add a short comment mentioning this since it's not immediately clear?

Reply:
Sounds good, added it.